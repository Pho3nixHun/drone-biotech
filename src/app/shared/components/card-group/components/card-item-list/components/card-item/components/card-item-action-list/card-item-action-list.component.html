<div
    class="flex flex-col xl:items-center xl:flex-row space-y-3 xl:space-y-0 xl:space-x-3"
>
    <ng-content select="button" />
</div>
