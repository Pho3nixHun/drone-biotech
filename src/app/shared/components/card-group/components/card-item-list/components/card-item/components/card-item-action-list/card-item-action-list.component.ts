import { Component } from '@angular/core';

@Component({
    selector: 'app-card-item-action-list',
    imports: [],
    templateUrl: './card-item-action-list.component.html',
})
export class CardItemActionListComponent {}
