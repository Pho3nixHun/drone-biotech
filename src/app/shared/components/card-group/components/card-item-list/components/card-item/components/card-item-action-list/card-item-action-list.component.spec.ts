import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CardItemActionListComponent } from './card-item-action-list.component';
import { Component } from '@angular/core';

@Component({
    imports: [CardItemActionListComponent],
    template: `
        <app-card-item-action-list>
            <button>Should be projected</button>
            <button>Should be projected</button>
            <p>Should not be projected</p>
        </app-card-item-action-list>
    `,
})
class TestHostComponent {}

describe('CardItemActionListComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should project the content and render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
