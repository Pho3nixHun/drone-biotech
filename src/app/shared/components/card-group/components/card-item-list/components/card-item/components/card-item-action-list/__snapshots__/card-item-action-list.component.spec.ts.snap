// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItemActionListComponent should project the content and render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-card-item-action-list>
    <div
      class="flex flex-col xl:items-center xl:flex-row space-y-3 xl:space-y-0 xl:space-x-3"
    >
      <button>
        Should be projected
      </button>
      <button>
        Should be projected
      </button>
    </div>
  </app-card-item-action-list>
</div>
`;
